docs/madr/decisions/039-transition-to-new-policies.md

```diff
@@ -44,12 +50,121 @@ spec:
   to: [...]
 ```
 
-it doesn't apply on the real proxies. Instead, by using a special endpoint we can get a list of jsonPatches that would be applied to the Envoy configuration
+it doesn't apply on the real proxies. Instead, by using Inspect API we can inspect what config dump would look like if the shadow policies was applied.
+
+### Changes in the Inspect API
+
+Introduce a new non-physical endpoint `/_snapshot` that returns a config dump of a DPP without the actual `/config_dump` call to Envoy.
+If we want to be able to compute diff on the client side we can't use `xds` endpoint as its response lightly diverges from what CP has in SnapshotCache.
+
+Add a query parameter `effect=shadow`:
+
+* `GET /meshes/{mesh}/dataplanes/{name}/_snapshot` -> `GET /meshes/{mesh}/dataplanes/{name}/_snapshot?effect=shadow`
+* `GET /meshes/{mesh}/dataplanes/{name}/_rules` -> `GET /meshes/{mesh}/dataplanes/{name}/_rules?effect=shadow`
+
+when `effect=shadow` is set, the response will take policies with `kuma.io/effect: shadow` label into account.
+Please note, the response DOES NOT yet contain the json patch diff. These endpoints are only useful when calculating the diff on the client side.
+
+Introduce a new non-physical endpoint `/_snapshot-diff`. This endpoint will return a list of changes in a JSON Patch format
+that shows what modifications are going to be applied to the config dump if we take into account shadow policies.
+
+We don't want to introduce `/_rules-diff` at this point, as `/_rules` API is rarely used without the client and heavily depends on the visualization in GUI.
+
+OpenAPI spec for new endpoints:
+
+```yaml
+openapi: 3.0.3
+info:
+  title: Shadow Policies Inspector
+  description: API for getting a diff in JSON Patch format as if shadow policies were applied
```

Comment:
the diff was supposed to be under `_snapshot-diff` right? not under `_snapshot`?

Reply:
right, I updated the description